File: tlx toki/review (python version)/Chapter_12/C_segitiga_bebek.py
x = [0] * 300
y = [0] * 300

def luas(a, b, c):
    return abs(0.5*(x[a] * (y[b] - y[c]) + x[b] * (y[c] - y[a]) + x[c] * (y[a] - y[b])))

n = int(input())
for i in range(n):
    x[i], y[i] = map(int, input().split())

res = 10**9
b = True
for i in range(n - 2):
    for j in range(i + 1, n - 1):
        for k in range(j + 1, n):
            t = luas(i, j, k)
            if t == 0:
                continue
            
            if t == res:
                b = True
            elif t < res:
                res = t
                b = False
if b:
    print(-1)
else:
    print(f"{res:.2f}")

File: CP/tlx toki/review (python version)/Chapter_12/C_segitiga_bebek.cpp
#include <bits/stdc++.h>
using namespace std;

double x[300], y[300];

double luas(int a, int b, int c){
    return abs(x[a]*(y[b]-y[c])+x[b]*(y[c]-y[a])+x[c]*(y[a]-y[b]))/2;
}
int main(){

    int n; cin >> n;
    for(int i = 0; i < n; i++) cin >> x[i] >> y[i];

    bool b = false;
    double res = 1e9;
    for(int i = 0; i < n - 2; i++)
        for(int j = i + 1; j < n - 1; j++)
            for(int k = j + 1; k < n; k++){
                double t = luas(i, j, k);
                if(t == 0) continue;

                if(t == res){
                    b = false;
                }else if(t < res){
                    b = true;
                    res = t;
                }
            }
    if(!b) cout << "-1";
    else cout << fixed << setprecision(2) << res;
}